### PR TITLE
Add support for completing shadows in super mixins

### PR DESCRIPTION
**Pull request by Minecrell**
Needs some polishing but seems to work fine so far.

I've also changed `PsiClass.mixinTargets` to be cached since it is called quite frequently.

### CODE REVIEW

**Comment by DemonWav** [comment]
4 /4 checks failed, gg.

**Comment by Minecrell** [comment]
@DemonWav Fixed

**Comment by Minecrell** [comment]
I added a hint when inserting shadows in super mixins:

screenshot_20170401_130826

File: src/main/kotlin/com/demonwav/mcdev/platform/mixin/util/TargetClass.kt [hunk 1]
```diff
@@ -10,54 +10,86 @@
...
     }?.filter {
         // Filter fields which are already in the Mixin class
-        psiClass.findFieldByName(it.name, checkBases) == null
+        psiClass.findFieldByName(it.name, false) == null
     }
 }
 
 @Contract(pure = true)
+fun findShadowTargets(psiClass: PsiClass, start: PsiClass, superMixin: Boolean): Sequence<ShadowTarget> {
+    return if (superMixin)
```
Comment (DemonWav) [inline src/main/kotlin/com/demonwav/mcdev/platform/mixin/util/TargetClass.kt]: _grumbles something about braces_

File: src/main/kotlin/com/demonwav/mcdev/platform/mixin/completion/MixinCompletionContributor.kt [hunk 2]
```diff
@@ -45,38 +44,56 @@ class MixinCompletionContributor : CompletionContributor() {
 
         // Check if completing inside Mixin class
         val psiClass = position.findContainingClass() ?: return
-        val targets = psiClass.mixinTargets.ifEmpty { return }
+        if (!psiClass.isMixin) return
```
Comment (DemonWav) [inline src/main/kotlin/com/demonwav/mcdev/platform/mixin/completion/MixinCompletionContributor.kt]: _grumbles something about braces_

File: src/main/kotlin/com/demonwav/mcdev/platform/mixin/action/GenerateOverwriteAction.kt [hunk 3]
```diff
@@ -13,7 +13,6 @@ package com.demonwav.mcdev.platform.mixin.action
 import com.demonwav.mcdev.platform.mixin.util.MixinConstants
 import com.demonwav.mcdev.platform.mixin.util.findMethods
 import com.demonwav.mcdev.platform.mixin.util.findSource
-import com.demonwav.mcdev.platform.mixin.util.mixinTargets
 import com.demonwav.mcdev.util.MinecraftFileTemplateGroupFactory.Companion.MIXIN_OVERWRITE_FALLBACK
 import com.demonwav.mcdev.util.findContainingClass
```
Comment (DemonWav) [inline src/main/kotlin/com/demonwav/mcdev/platform/mixin/action/GenerateOverwriteAction.kt]: `.Companion`

File: src/main/kotlin/com/demonwav/mcdev/platform/mixin/completion/MixinCompletionContributor.kt (hunk 2 above)
Comment (kashike) [inline src/main/kotlin/com/demonwav/mcdev/platform/mixin/completion/MixinCompletionContributor.kt]: b

**Pull request merged**